### PR TITLE
Added Sqrt() for Fraction

### PR DESCRIPTION
**Pull request by MadsKirkFoged**
This is my attempt to create the sqrt() function for Fractions.

I hope it is okay that I have created it as an Extension Method, I like this:
`Fraction.One.Sqrt()`
over 
`Fraction.Sqrt(Fraction.One)`

I have never used FluentAssertions test system, but I hope what I have created will work for you.

What do you think?

### CODE REVIEW

**Comment by danm-de** [comment]
Hey, thanks. 
I will take a closer look at it over the weekend

**Comment by MadsKirkFoged** [comment]
Just added a way for users to set their desired accuracy

File: tests/Fractions.Tests/FractionSpecs/Sqrt/Method_Sqrt.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            SqrtCompareToDouble(9999999855487).Should().BeTrue();
+            SqrtCompareToDouble(99999998554865557).Should().BeTrue();
+            SqrtCompareToDouble(Math.PI).Should().BeTrue();
+
+        }
+
+
+
+
+
```
Comment (danm-de) [inline tests/Fractions.Tests/FractionSpecs/Sqrt/Method_Sqrt.cs:49]: Yeah, the tests... *Fractions* started as private project. Nearly every single line was written in German language 🥲. I am sorry for not writing proper tests in the first place. However, the use of TestCases makes more verbose tests:

image

**CAUTION** SUGGESTION REQUIRES A NEW USING `using System.Collections.Generic;` 

```suggestion
        private static IEnumerable<TestCaseData> TestCases {
            get {
                yield return new TestCaseData(0.001);
                yield return new TestCaseData(5);
                yield return new TestCaseData(4.54854751);
                yield return new TestCaseData(9999999855487);
                yield return new TestCaseData(99999998554865557);
                yield return new TestCaseData(Math.PI);
            }
        }

        [Test,TestCaseSource(nameof(TestCases))]
        public void Should_the_result_be_equal_to_Math_Sqrt(double value) {
            var expected = Math.Sqrt(value);
            var actual = Fraction.FromDouble(value).Sqrt();
            actual.ToDouble().Should().Be(expected);
        }```

**Review by danm-de — CHANGES_REQUESTED** [review]

File: src/Fractions/Fraction.Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            }
+
+
+            Fraction oldGuess;
+            var newGuess = Fraction.Zero;
+            var tolerance = new Fraction(BigInteger.One, BigInteger.Pow(new BigInteger(10), numberOFDecimalPlaceAccuracy));
+
+
+            //Using Math.Sqrt to get a good starting guess
+            var guessDouble = Math.Sqrt((double)x);
```
[the remark this answers is not in the document]
Comment (MadsKirkFoged) [reply src/Fractions/Fraction.Extensions.cs:29]: I tested it - It cant overflow, it just set the double to INF which we then checks for

**Comment by MadsKirkFoged** [comment]
I think I have fixed your request! Let me know if I missed something or you find something else!

**Pull request merged**